internal/clearingway/guild.go

```diff
@@ -246,3 +280,61 @@ func (g *Guild) IsProgEnabled() bool {
 
 	return false
 }
+
+func (g *Guild) InitDiscordMenu() {
+	g.ComponentsHandlers = make(map[string]func(s *discordgo.Session, i *discordgo.InteractionCreate))
+
+	menuButtons := []discordgo.MessageComponent{}
+
+	// Verify Clears
+	dataMenuVerify := g.Menus.Defaults[MenuVerify]
+	menuButtons = append(menuButtons, &discordgo.Button{
+		Label: dataMenuVerify.Title,
+		Style: discordgo.SuccessButton,
+		Disabled: false,
+		CustomID: MenuVerify,
+	})
+	g.ComponentsHandlers[MenuVerify] = nil  // placeholder
+
+	// Remove Roles
+	dataMenuRemove := g.Menus.Defaults[MenuRemove]
+	menuButtons = append(menuButtons, &discordgo.Button{
+		Label: dataMenuRemove.Title,
+		Style: discordgo.DangerButton,
+		Disabled: false,
+		CustomID: MenuRemove,
+	})
+	g.ComponentsHandlers[MenuRemove] = nil  // placeholder
+
+	dataMenuMain := g.Menus.Defaults[MenuMain]
+	menuMessage := &discordgo.MessageSend{
+		Embeds: []*discordgo.MessageEmbed{
+			{
+				Title: dataMenuMain.Title,
+				Description: dataMenuMain.Description,
+			},
+		},
+		Components: []discordgo.MessageComponent{
+			discordgo.ActionsRow{
+				Components: menuButtons,
+			},
+		},
+	}
+
+	// create a function that sends the menuMessage generated above
+	// specific to each guild
+	g.ComponentsHandlers[MenuMain] = func(s *discordgo.Session, i *discordgo.InteractionCreate) {
+		_, err := s.ChannelMessageSendComplex(i.ChannelID, menuMessage)
+		if err != nil {
+			fmt.Printf("Error sending Discord message: %v\n", err)
+			return
+		}
+
+		err = discord.StartInteraction(s, i.Interaction, "Sent menu message.")
+		if err != nil {
+			fmt.Printf("Error sending Discord message: %v\n", err)
+			return
+		}
+	}
```

Comment:
this syntax errs on the side of javascript, and i'm not sure if we want to go this route. might be better to just make a separate helper function

Reply:
I believe this ties into https://github.com/Veraticus/clearingway/pull/32#discussion_r1729503641 so I didn't change it much till further discussion. For now I just created a new helper function that essentially returns the same thing but with slightly improved readability.